### ISSUE TITLE
Clarify how users can use the search endpoint 

### ISSUE THREAD

**Issue by emmambd** [MobilityData/mobility-feed-api#442]
### Describe the problem

[In the current Swagger documentation](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/search/searchFeeds), it isn't clear 

- What types of operations the search endpoint supports (e.g AND, OR, etc)
- The specific types of entities that are included in the search endpoint 

Users need to know what queries they can use and what is covered. 

### Suggest a fix or an alternative

Example of what this can look like from Crunchbase's API in[ the Key Reminders section](https://data.crunchbase.com/docs/examples-search-api)

- Rename search_query parameter description from 
`General search query to match against various fields of the entities.` to `General search query to match against transit provider, location, and feed name.`
- Clarify which operations are supported in the search endpoint description 
- Include a search query example in the parameter field

### Considerations

@jcpitre had some questions regarding how appropriate match works. Should be discussed:

`Linee Urbane di Ostuni`
Returned mdb-2020 that has: "feed_name": "Linee Urbane di Ostuni"
`Linee Urbane di Ostun` (note the missing i at the end)
Returned nothing
`Line Urbane di Ostuni` (note the missing e on the first word)
Returned mdb-2020

**Comment by jcpitre** [comment]
Tasks:

- [ ] Document what we have + examples

**Comment by davidgamez** [comment]
The current search endpoint implementation is based on Postgresql ts_vector functions, [more info here](https://github.com/MobilityData/mobility-feed-api/blob/1b9bff04b58abbc327e2eb2f527a49220b17e38b/api/src/feeds/impl/search_api_impl.py#L26). In particularly the `plainto_tsquery`, [official documentation](https://www.postgresql.org/docs/current/textsearch-controls.html). 

As official documentation:
----
```
`plainto_tsquery`transforms the unformatted text querytext to a tsquery value. The text is parsed and normalized much as for to_tsvector, then the & (AND) tsquery operator is inserted between surviving words.
```
Example:
```
SELECT plainto_tsquery('english', 'The Fat Rats');
 plainto_tsquery
-----------------
 'fat' & 'rat'
```
Note that plainto_tsquery will not recognize tsquery operators, weight labels, or prefix-match labels in its input:
```
SELECT plainto_tsquery('english', 'The Fat & Rats:C');
   plainto_tsquery
---------------------
 'fat' & 'rat' & 'c'
```
----

This means:
- Operators are not currently supported. Operators are ignored as stop-words.
- The search will match all the lexemes with an AND operator. So, all lexemes must be present in the document.
- Our current implementation only creates English lexemes, as the API needs a straightforward way to get the user's language. 
- The order of the words is not relevant for matching. The query `New York` should give you the same results as `York New`.


## Explaining examples

Query: Linee Urbane di Ostuni
Returned mdb-2020 that has: "feed_name": "Linee Urbane di Ostuni"
Explanation: The query contains the complete text

Query: Linee Urbane di Ostun (note the missing i at the end)
Returned nothing
Explanation: "Ostun" is not an English term. No lexemes are derived from the word Ostun( nor Ost neither Ostu ), so it's missing from the document.

Query: Line Urbane di Ostuni (note the missing e on the first word)
Returned mdb-2020
Explanation: Line is an English word; this matches the Linee with Line; all other words are in the document.

## Areas of improvements 

- Support more languages:
   - We can add more parsing languages. This will require adding additional columns to the materialized view and possible changes in how queries are made. A performance review should be done as this will increase the processing time of the query and ranking calculation
- Support operators(not complete full-text search):
   - Within the PostgreSQL we can use websearch_to_tsquery function. As per documentation
   `websearch_to_tsquery creates a tsquery value from querytext using an alternative syntax in which simple unformatted text is a valid query. Unlike plainto_tsquery and phraseto_tsquery, it also recognizes certain operators.` This allows the user to have more control with the search operations; [official documentation](https://www.postgresql.org/docs/current/textsearch-controls.html). 
 
- Implementing a fully full-text search solution:
  - This was considered when designing the MVP but discarded as it requires infrastructure changes and additional cost. We will keep monitoring requirements and the size of the database to implement this type of solution in a timely manner. Full discussion here, #370

**Comment by emmambd** [comment]
Thanks for this detailed breakdown @davidgamez! I'd like to do some search testing with the spec team once @indraneel is done the search functionality, and we can decide what modifications are critical at that point before launch. I think adding more languages will be critical, but I don't foresee any other needed changes at this point.

**Comment by emmambd** [comment]
Closed in #454
